### DISCUSSION TITLE
"You don't have permission to access this." - uploading file(image)

### DISCUSSION THREAD

**Discussion by Velpre** [directus/directus#16229] [Archive]
Hi

I am getting this error while i am trying to create new user. 
"You don't have permission to access this."

Error occurs because I am trying to upload a image while creating new user.

Function I use for file(image) upload works fine if the user is already logged in. I already use that method elsewhere while the user is logged in and then it works fine.

How can I use it while registering a new user?

the method i am using for uploading images is:
```
const newFormData = new FormData();
newFormData.append("file", imageBlob) //Blob is already made
await directus.files.createOne(newFormData)
```

I have tested the register user function with already predefined files(images) in directus, it works fine. Problem I get occurs when I try to upload newFormData.

How can i upload new file while creating a new user?

Thank you in advance

**Comment by raphourbe** [comment]
Hi,

When you say that you try to create a new user, what is the Role used to do the creation? If it is the user, and he is not logged in, it means he is in the Public role.
What are the Public role permissions regarding **Directus File** in the system collections? I would say they need to be allowed to create at least.

Hope this helps :)

**Reply by Velpre** [reply]
That helped me a lot. I forgot that we need to manage permissions for public users as well. 
Thank you very much for your fast answer :)
